### ISSUE TITLE
Check the consistency of the environment

### ISSUE THREAD

**Issue by wush978** [RBigData/pbdMPI#53]
Hi,

Thanks for your great work. I have used the pbdMPI for two years.

Under cluster mode, sometimes the inconsistent environment will cause some errors. Sometimes, it will take user a lot of time to debug because the error message usually does not indicate the inconsistency of the environment.
It would be great if there is a helper function to check the consistency of the environment such as package version, locale, and timezone and warn the user which part and which node is inconsistent.

**Comment by snoweye** [comment]
It will be good to have a useful example of the helper function. The request is unclear to me, for example, what environment is needed for checking. When there is inconsistent, the code must be stop. In such a case, I do not see a helper function can be useful except casting messages which were already done by comm.cat() and comm.print(). When there is different package version or else, it is system administration problems.

**Comment by wush978** [comment]
In my case, I use the `sessionInfo` to print the locales and package versions. However, the message becomes too many for human if there are many nodes in the cluster and I need to write my own checking functions to ensure the consistency.

I agree that different package version, locales, time zones or other environment settings are the problem of system administration. However, A convenient helper function will help the users to discover such problem more easily. And such checking should be done before every complicated application based on pbdMPI to prevent the occurrence of any obscure errors. Therefore, I think it will be great if pbdMPI could provide such kind of helper function.

**Comment by snoweye** [comment]
'?comm.allcommon' should be sufficient, isn't it?

**Comment by wush978** [comment]
Yes, thanks!
